app/models/doc_version.rb

```diff
@@ -24,21 +24,18 @@ class DocVersion < ApplicationRecord
   # 2: 8 - (add + sub)
   def diff(doc_version)
     begin
-      to = self.doc.plain.split("\n")
-      from = doc_version.doc.plain.split("\n")
-      total = adds = mins = 0
-      diff = Diff::LCS.diff(to, from)
-      diff.first.each do |change|
-        adds += 1 if change.action == "+"
-        mins += 1 if change.action == "-"
-        total += 1
-      end
-      if total > 8
+      diff_out = Diffy::Diff.new(self.doc.plain, doc_version.doc.plain)
+      first_chars=diff_out.to_s.gsub(/(.)[^\n]*\n/, '\1')
```

Comment:
This will also create a big array of ruby strings, though it is now proportional to the size of the diff rather than the size of the file. They are the same size worst-case, but in practice the diff will be much smaller. So it probably doesn't matter too much, but I suspect we could just count `/^+/` and `/^-/` directly with a regex. Or maybe just using the enumerable interface, like:

```ruby
Diff::Diff.new(self.doc.plain, doc_version.doc.plain).each do |line|
  case line
  when /^\+/ then adds += 1
  when /^-/ then mins += 1
  end
end
```

(which I stole from the diffy README).

Reply:
I don't understand your comment: "This will also create a big array of ruby strings". The two line of my changes only convert one big string to another one. See in Diffy:

https://github.com/samg/diffy/blob/ff2103d718c581964769d345ae23596bf8242213/lib/diffy/diff.rb#L57

The idea here is to not resort to a loop in ruby, because we know that ruby is so slow. So I use regexes and native primitives so that the processing is not interpreted.

Reply:
Sorry, you're right; I was mis-reading the ruby code and thinking you were creating an array of matched expressions, not a string of them. So yeah, I think this is avoiding as much ruby-level logic as possible.